compiler/rustc_type_ir/src/solve/mod.rs

```diff
@@ -177,8 +177,9 @@ pub enum BuiltinImplSource {
     /// A built-in implementation of `Upcast` for trait objects to other trait objects.
     ///
     /// This can be removed when `feature(dyn_upcasting)` is stabilized, since we only
-    /// use it to detect when upcasting traits in hir typeck.
-    TraitUpcasting,
+    /// use it to detect when upcasting traits in hir typeck. The index is only used
+    /// for winnowing.
+    TraitUpcasting(usize),
```

Comment:
Is this still true? Can this be removed?...

Reply:
True, can't remove it now, not that it really matters in the first place xd

Reply:
Should I include the removal in stabilization PR or is this a future cleanup style of thing? I'm honestly confused how this can be removed, if it is used for winnowing (waffle silly).

Reply:
I wrote "can't" remove it in my comment 😸 we definitely do need to keep it around for winnowing, yep.

Specifically, we need to keep it around for the same reason we have `BuiltinImplSource::Object`. You can adjust the comment in the stabilization report to remove the "This can be removed when [...]" part, and only keep the sentence explaining what the index is for.

Reply:
Oh. I interpreted your comment as "[this sentence is still] True, can't remove it [for] now" 😸 

Okay, I'll do that :3